src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/PublicKeySignatureWriter.java

```diff
@@ -40,25 +42,30 @@ public String asString(PublicKey key) {
     }
 
     public String asString(DSAPublicKey key) {
-        TypesWriter tw = new TypesWriter();
-        tw.writeString("ssh-dss");
-        DSAParams p = key.getParams();
-        tw.writeMPInt(p.getP());
-        tw.writeMPInt(p.getQ());
-        tw.writeMPInt(p.getG());
-        tw.writeMPInt(key.getY());
-        return encode(tw);
+        try {
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
+            DSAParams keyParams = Objects.requireNonNull(key.getParams(), "No DSA params available");
+            KeyEncodeHelper.encodeString(output, "ssh-dss", StandardCharsets.UTF_8);
+            KeyEncodeHelper.encodeBigInt(output, keyParams.getP());
+            KeyEncodeHelper.encodeBigInt(output, keyParams.getQ());
+            KeyEncodeHelper.encodeBigInt(output, keyParams.getG());
+            KeyEncodeHelper.encodeBigInt(output, key.getY());
+            return new String(output.toByteArray(), StandardCharsets.UTF_8);
+        } catch(IOException e) {
+            throw new PublicKeySignatureWriterException(e);
```

Comment:
Not sure what the purpose of this wrapper exception is.

Reply:
the current methods do not throw an exception (only runtime if them happens), so in order to do not change the current behavior I've transformed the IOException to a RunTimeException because any kind of error should be fatal.

Reply:
Oh, missed that this was an unchecked exception type. Fine, or you can just wrap as `RuntimeException`.

Reply:
I preferred to use a named exception to allow catch it if it is needed in some place, it is only a silly class and can worth